--- a/ndpi-netfilter/src/main.c	2023-12-22 18:09:04.107421950 +0100
+++ b/ndpi-netfilter/src/main.c	2023-12-22 18:10:11.038240353 +0100
@@ -103,13 +103,9 @@
 static char debug_name[]="debug";
 static char risk_name[]="risks";
 
+#if LINUX_VERSION_CODE > KERNEL_VERSION(5,19,0)
-#ifdef CONFIG_NF_CONNTRACK_DESTROY_HOOK
 #define USE_NF_CONNTRACK_DESTROY_HOOK
-#elif LINUX_VERSION_CODE > KERNEL_VERSION(5,19,0)
-#ifndef USE_LIVEPATCH
-#define USE_LIVEPATCH
-#endif
-#endif
+#endif
 
 #ifdef USE_LIVEPATCH
 #if IS_ENABLED(CONFIG_LIVEPATCH)
