--- a/ndpi-netfilter/src/main.c	2023-12-22 18:46:47.999596865 +0100
+++ b/ndpi-netfilter/src/main.c	2023-12-22 18:48:52.981393673 +0100
@@ -2840,7 +2840,11 @@
 #if LINUX_VERSION_CODE < KERNEL_VERSION(4, 8, 0)
 	net->ct.label_words = n->labels_word;
 #endif
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 7, 0)
 	net->ct.labels_used--;
+#else
+	atomic_dec_return_relaxed(&net->ct.labels_used);
+#endif
 
 #if   LINUX_VERSION_CODE >= KERNEL_VERSION(5, 19, 0)
 	struct nf_ct_iter_data iter_data = {
@@ -3138,7 +3142,11 @@
 		n->labels_word = ACCESS_ONCE(net->ct.label_words);
 		net->ct.label_words = 2;
 #endif
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 7, 0)
 		net->ct.labels_used++;
+#else
+		atomic_inc_return_relaxed(&net->ct.labels_used);
+#endif
 #endif
 		if( ndpi_enable_flow &&
 		    nf_register_net_hooks(net, nf_nat_ipv4_ops,
