--- a/src/include/ndpi_define.h.in	2023-12-25 08:45:56.692170223 +0100
+++ b/src/include/ndpi_define.h.in	2023-12-25 08:46:11.595904820 +0100
@@ -326,7 +326,6 @@
 #define get_u_int16_t(X,O)  (*(u_int16_t *)((&(((u_int8_t *)X)[O]))))
 #define get_u_int32_t(X,O)  (*(u_int32_t *)((&(((u_int8_t *)X)[O]))))
 #if defined(__arm__)
-#include <stdint.h>
 static inline uint64_t get_u_int64_t(const uint8_t* X, int O)
 {
   uint64_t tmp;
