--- a/src/lib/protocols/whatsapp.c.anc	2018-10-22 09:32:53.953147328 +0200
+++ b/src/lib/protocols/whatsapp.c	2018-10-22 09:33:07.560968148 +0200
@@ -32,7 +32,7 @@
   };
 
   NDPI_LOG_DBG(ndpi_struct, "search WhatsApp\n");
-
+/*
   if(flow->l4.tcp.wa_matched_so_far == 0) {
     if(memcmp(packet->payload, whatsapp_sequence, packet->payload_packet_len)) {
       NDPI_EXCLUDE_PROTO(ndpi_struct, flow);
@@ -55,7 +55,7 @@
     NDPI_LOG_INFO(ndpi_struct, "found WhatsApp\n");
     ndpi_set_detected_protocol(ndpi_struct, flow, NDPI_PROTOCOL_WHATSAPP, NDPI_PROTOCOL_UNKNOWN);
   }
-
+*/
   NDPI_EXCLUDE_PROTO(ndpi_struct, flow);
 }
 
