Index: sierra-cm-SLQS03.03.10.bin/SampleApps/Makefile
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ sierra-cm-SLQS03.03.10.bin/SampleApps/Makefile	2016-05-24 21:15:52.585360952 +0530
@@ -0,0 +1,9 @@
+FOLDERS=QCA_Connection_Manager
+
+.PHONY: clean
+
+all:
+	@for folder in ${FOLDERS}; do ${MAKE} -C $${folder} CPU="arm" || exit 1; done
+
+clean:
+	@for folder in ${FOLDERS}; do ${MAKE} -C $${folder} clean || exit 1; done
Index: sierra-cm-SLQS03.03.10.bin/SampleApps/SampleApps.mak
===================================================================
--- sierra-cm-SLQS03.03.10.bin.orig/SampleApps/SampleApps.mak	2015-07-09 08:44:56.000000000 +0530
+++ sierra-cm-SLQS03.03.10.bin/SampleApps/SampleApps.mak	2016-05-24 21:20:02.954616238 +0530
@@ -43,8 +43,8 @@
 VPATH   =
 SRCDIR  = src/
 ODIR    = obj/$(CPU)/
-LDIR    = ../../build/lib/$(CPU)/ -static -lrt -lpthread
+LDIR    = ../build/lib/$(CPU)/ -static -lrt -lpthread
 BINDIR  = bin/
 
 DEPS    = $(IDIR)SWIWWANCMAPI.h
-LIBS    = -l slqsqmi -l slqsapi -l slqscore -L $(LDIR) 
+LIBS    = -l slqsqmi -l slqsapi -l slqscore -L $(LDIR) $(MYFLAGS)
Index: sierra-cm-SLQS03.03.10.bin/pkgs/slqscompile.mak
===================================================================
--- sierra-cm-SLQS03.03.10.bin.orig/pkgs/slqscompile.mak	2015-07-09 08:44:53.000000000 +0530
+++ sierra-cm-SLQS03.03.10.bin/pkgs/slqscompile.mak	2016-05-24 21:18:34.838174071 +0530
@@ -7,7 +7,7 @@
   QMI_BUILD_BASE:=$(shell until [ -d pkgs ]; do cd ..; done; pwd)
 endif
 SLQSHOSTTYPE := $(shell uname -m)
-CFLAGS       := -c -Wall -Werror -Wextra
+CFLAGS       := -c -Wall -Wextra
 INCLUDE      := -I.. $(QMI_INCLUDE)
 LDFLAGS      := -static -lrt -lpthread
 DEPSDIR      := ./dep
@@ -84,10 +84,12 @@
 # will simply fail.
 #-------------------------------------------------------------------------------
 ifeq ($(CPU),arm)
+  ifndef MYFLAGS
   CROSS_COMPILE := $(HOME)/toolchain/arm/bin/arm-none-linux-gnueabi-
   INCLUDE       += -I$(HOME)/toolchain/arm/arm-none-linux-gnueabi/libc/usr/include
   LDFLAGS       += -L$(HOME)/toolchain/arm/lib/gcc/arm-none-linux-gnueabi/4.2.0/
   CFLAGS        += -march=armv5te
+  endif
 endif
 # this is for backward compatibility
 ifeq ($(CPU),arm9)
