diff --git a/qca-cnss-daemon/patches/Makefile b/qca-cnss-daemon/patches/Makefile
new file mode 100644
index 0000000..f413dd8
--- /dev/null
+++ b/Makefile
@@ -0,0 +1,33 @@
+INCLUDE_DIR += -I. -I$(STAGING_DIR)/usr/include/libnl \
+		    -I$(STAGING_DIR)/usr/include/libnl3
+
+FLAGS := -fPIC -Wall -g -DICNSS_QMI -D__USE_GNU -DCONFIG_DEBUG_SYSLOG \
+	-DCONFIG_DEBUG -DCONFIG_DEBUG_FILE -DCONFIG_IPC_LOGGING \
+	-DCONFIG_NO_DMS -D_GNU_SOURCE \
+	-DCONFIG_NO_DATA_PATH_EXTNS \
+	-DCONFIG_NO_PM -DCONFIG_CNSS_USER\
+	-DCONFIG_RECORD_DAEMON_QMI_LOG -DCONFIG_WLAN_MSG_SVC \
+	-DDISABLE_CLD80211_LIB -D__NON_ANDROID -DIPQ
+LIBS := -lpthread -lnl-genl-3 -lnl-3
+LDFLAGS := -L/usr/lib -lqmi_common_so -lqmi_qrtr_cci \
+	-pie
+
+CFLAGS += -fpie
+
+cnssdaemon := wlan_firmware_service_v01.o \
+	cnss_plat.o wlfw_qmi_client.o \
+	debug.o \
+	main.o nl_loop.o cnss_user.o cnss_genl.o
+
+cnsscli := cnss_cli.o
+%.o: %.c
+	$(CC) -c -o $@ $(INCLUDE_DIR) $(FLAGS) $(LIBS) $(LDFLAGS) $(CFLAGS) $<
+
+all: $(cnssdaemon) $(cnsscli)
+	$(CC) -o cnssdaemon $(cnssdaemon) $(INCLUDE_DIR) $(FLAGS) $(LIBS) $(LDFLAGS) $(CFLAGS)
+	$(CC) -o cnsscli $(cnsscli) $(INCLUDE_DIR) $(LIBS) $(LDFLAGS) $(CFLAGS)
+	mkdir -p lib/
+	mv cnssdaemon lib/
+	mv cnsscli lib/
+clean:
+	rm -rf lib/
+	rm -rf *.o
+
