--- a/po/Makefile.in.in 2020-07-13 04:16:44.000000000 +0800
+++ b/po/Makefile.in.in 2021-02-17 18:04:44.881887192 +0800
@@ -8,8 +8,8 @@
 # Please note that the actual code of GNU gettext is covered by the GNU
 # General Public License and is *not* in the public domain.
 #
-# Origin: gettext-0.18
-GETTEXT_MACRO_VERSION = 0.18
+# Origin: gettext-0.20
+GETTEXT_MACRO_VERSION = 0.20

 PACKAGE = @PACKAGE@
 VERSION = @VERSION@