--- a/src/flowlabel.h	2021-06-24 13:26:33.142463630 +0200
+++ b/src/flowlabel.h	2021-06-24 13:27:45.669235179 +0200
@@ -37,21 +37,21 @@
    conflicts with "netinet/in.h" .
 */
 
-#ifndef __ANDROID__
+#ifndef _LINUX_IN6_H
 struct in6_flowlabel_req
 {
     struct in6_addr flr_dst;
     __u32   flr_label;
     __u8    flr_action;
     __u8    flr_share;
     __u16   flr_flags;
     __u16   flr_expires;
     __u16   flr_linger;
     __u32   __flr_pad;
     /* Options in format of IPV6_PKTOPTIONS */
 };
 #endif
 
 #define IPV6_FL_A_GET           0
 #define IPV6_FL_A_PUT           1
 #define IPV6_FL_A_RENEW         2
