diff --git a/cnssdiag/Makefile b/cnssdiag/Makefile
index d1d340a..3923cbf 100644
--- a/cnssdiag/Makefile
+++ b/cnssdiag/Makefile
@@ -1,9 +1,8 @@
-CC := $(ATH_CROSS_COMPILE_TYPE)gcc
-TARGET_TYPE ?= AR9888
-TARGET_VERS ?= v2
+CC = gcc
+AR = ar
 
 all:
-	$(CC) -g3 -Wall -Werror -Wno-unusedparameter \
+	$(CC) -g3 -Wall -Werror $(CFLAGS) $(LDFLAGS) \
 		cld-fwlog-netlink.c parser.c nan-parser.c cld-diag-parser.c -o cnss_diag
 
 clean:
