--- a/pinctrl/pinctrl.c	2024-07-02 14:14:31.113459219 +0200
+++ b/pinctrl/pinctrl.c	2024-07-02 14:14:44.533241981 +0200
@@ -220,7 +220,7 @@
     if (!gpio_num_is_valid(gpio))
         return 1;
 
-    poll_gpios = reallocarray(poll_gpios, num_poll_gpios + 1,
+    poll_gpios = realloc(poll_gpios, num_poll_gpios + 1 *
                               sizeof(*poll_gpios));
     new_gpio = &poll_gpios[num_poll_gpios];
     new_gpio->num = num;
