diff --git a/src/http.c b/src/http.c
index 45d7281..99f9cfd 100644
--- a/src/http.c
+++ b/src/http.c
@@ -88,7 +88,7 @@ sock_t httpPut(char* pAddress, int pPort, char* pRequest, unsigned long contentS
 	 Leaving it uninitialized gives us random high value.*/
 	sprintf(buffer, "POST %s HTTP/1.1\r\n"
 			"Host: %s\r\n"
-			"User-Agent: SPEEDTEST_CLIENT\r\n"
+			"User-Agent: OMR_SPEEDTEST_CLIENT\r\n"
 			"Content-Type: application/x-www-form-urlencoded\r\n"
 			"Connection: keep-alive\r\n"
 			"Content-Length: %lu\r\n"
@@ -131,7 +131,7 @@ sock_t httpGet(char* pAddress, int pPort, char* pRequest, int ssl)
 
 	sprintf(buffer, "GET %s HTTP/1.1\r\n"
 			"Host: %s\r\n"
-			"User-Agent: SPEEDTEST_CLIENT\r\n"
+			"User-Agent: OMR_SPEEDTEST_CLIENT\r\n"
 			"Connection: close\r\n"
 			"\r\n", pRequest, pAddress);
 
