--- a/extensions/LUA/controller.h	2024-09-26 16:49:17.099012595 +0200
+++ b/extensions/LUA/controller.h	2024-09-26 16:49:39.098706793 +0200
@@ -244,6 +244,9 @@
 int get_string_generic(lua_State *L);
 int get_byte_generic_str(lua_State *L);
 struct field_changes * get_allocated_field_changes(lua_State *L, int nr_of_fields);
+int32_t get_header_size(struct protocol_buf * prot_buf);
+struct field_changes * tftp_get_field_changes(lua_State *L, lua_packet_segment * seg);
+void free_dynamic_prot_buf(struct protocol_buf * prot_buf);
 
 /* only used by the dynamic prot buf subsystem */
 #define MAX_NR_OF_DYN_PROT_BUFS 16
