--- a/extensions/xt_ipp2p.c	2024-10-08 14:14:40.759989875 +0200
+++ b/extensions/xt_ipp2p.c	2024-10-08 14:16:04.042851316 +0200
@@ -6,7 +6,11 @@
 #include <linux/netfilter_ipv4/ip_tables.h>
 #include <net/tcp.h>
 #include <net/udp.h>
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 11, 0)
 #include <asm/unaligned.h>
+#else
+#include <linux/unaligned.h>
+#endif
 #include "xt_ipp2p.h"
 #include "compat_xtables.h"
 
