--- a/extensions/LUA/prot_buf_ip.c	2023-10-20 16:55:00.658340536 +0200
+++ b/extensions/LUA/prot_buf_ip.c	2023-10-20 16:55:11.030165998 +0200
@@ -17,7 +17,6 @@
  */
 
 #if defined(__KERNEL__)
-	#include <net/checksum.h>
 	#include <net/tcp.h>
 #endif
 
--- a/extensions/LUA/prot_buf_tcp.c	2023-10-20 17:05:44.507457150 +0200
+++ v/extensions/LUA/prot_buf_tcp.c	2023-10-20 17:06:08.907041173 +0200
@@ -17,7 +17,6 @@
  */
 
 #if defined(__KERNEL__)
-	#include <net/checksum.h>
 	#include <net/tcp.h>
 #endif
 #include "controller.h"
--- a/extensions/LUA/prot_buf_udp.c	2023-10-20 17:05:54.411288304 +0200
+++ b/extensions/LUA/prot_buf_udp.c	2023-10-20 17:06:15.758924360 +0200
@@ -16,10 +16,9 @@
  *	along with this program; if not, see <http://www.gnu.org/licenses/>.
  */
 
 #if defined(__KERNEL__)
-	#include <net/checksum.h>
 #endif
 
 #include "controller.h"
 
 
