--- a/src/config.c	2018-07-01 19:02:09.192210482 +0200
+++ b/src/config.c	2018-07-01 19:03:13.263404635 +0200
@@ -302,9 +302,11 @@
                         config, lastSection, "remoteport", &dstport, NULL,
                         "No remote port specified.\n", 1);
                 }
-
-                bindaddr=ip_from_if(bindaddr);
                 
+                if (bindaddr)
+                {
+                    bindaddr=ip_from_if(bindaddr);
+                }
                 
                 _conf_set_uint_from_conf(
                     config, lastSection, "bandwidth_upload", &bwlimit, 0,
