--- a/src/mlvpn.c	2018-08-25 07:39:53.625002006 +0200
+++ b/src/mlvpn.c	2018-08-25 07:40:35.696481066 +0200
@@ -1019,7 +1019,7 @@
 #endif
     if (*t->binddev) {
       memset(&ifr, 0, sizeof(ifr));
-      snprintf(ifr.ifr_name, sizeof(ifr.ifr_name) - 1, t->binddev);
+      snprintf(ifr.ifr_name, sizeof(ifr.ifr_name) - 1, " on %s", t->binddev);
       if (setsockopt(fd, SOL_SOCKET, SO_BINDTODEVICE, (void *)&ifr, sizeof(ifr)) < 0) {
         log_warn(NULL, "failed to bind on interface %s", t->binddev);
       }
