--- a/system-linux.c	2023-09-13 10:44:22.163111635 +0200
+++ b/system-linux.c	2023-09-13 10:44:47.562696256 +0200
@@ -2720,7 +2720,9 @@
 system_if_dump_info(struct device *dev, struct blob_buf *b)
 {
 	__u32 *supported, *advertising, *lp_advertising;
-	bool rx_pause, tx_pause, pause_autoneg;
+	bool rx_pause = false;
+	bool tx_pause = false;
+	bool pause_autoneg = false;
 	struct {
 		struct ethtool_link_settings req;
 		__u32 link_mode_data[3 * 127];
