diff -Nur a/scripts/netifd-wireless.sh netifd-2019-08-05-5e02f944/scripts/netifd-wireless.sh
--- a/scripts/netifd-wireless.sh	2020-05-01 14:55:34.404718599 -0700
+++ netifd-2019-08-05-5e02f944/scripts/netifd-wireless.sh	2020-05-01 15:00:20.171171673 -0700
@@ -204,9 +204,6 @@
 		*gcmp) wpa_cipher="GCMP";;
 	esac
 
-	# Standlone TKIP is no longer allowed
-	[ "$wpa_cipher" = "TKIP" ] && wpa_cipher="CCMP TKIP"
-
 	# Examples:
 	# psk-mixed/tkip    => WPA1+2 PSK, TKIP
 	# wpa-psk2/tkip+aes => WPA2 PSK, CCMP+TKIP
@@ -227,12 +224,17 @@
 			wpa=1
 		;;
 		*)
-			# TKIP alone is now prohibited by WFA so the only
-			# combination left must be CCMP+TKIP (wpa=3)
-			wpa=3
-			wpa_cipher="CCMP TKIP"
+			wpa=0
+			wpa_cipher=
 		;;
 	esac
+	# Standlone TKIP is no longer allowed
+	# TKIP alone is now prohibited by WFA so the only
+	# combination left must be CCMP+TKIP (wpa=3)
+	[ "$wpa_pairwise" = "TKIP" ] && {
+		wpa=3
+		wpa_cipher="CCMP TKIP"
+	}
 	wpa_pairwise="$wpa_cipher"
 
 	case "$encryption" in
