diff -Nur a/scripts/netifd-wireless.sh netifd-2019-08-05-5e02f944/scripts/netifd-wireless.sh
--- a/scripts/netifd-wireless.sh	2020-05-01 15:05:45.809410514 -0700
+++ netifd-2019-08-05-5e02f944/scripts/netifd-wireless.sh	2020-05-01 15:08:44.124446556 -0700
@@ -197,6 +197,8 @@
 	auth_type=none
 	wpa=0
 	wpa_cipher=
+	eapol_key_index_workaround=0
+
 	case "$encryption" in
 		*tkip+aes|*tkip+ccmp|*aes+tkip|*ccmp+tkip) wpa_cipher="CCMP TKIP";;
 		*aes|*ccmp| psk2 | wpa2) wpa_cipher="CCMP";;
@@ -256,8 +258,14 @@
 		*psk*)
 			auth_type=psk
 		;;
-		*wpa*|*8021x*)
+		*wpa*)
+			auth_type=eap
+			eapol_key_index_workaround=1
+		;;
+		*8021x*)
 			auth_type=eap
+			eapol_version=2
+			eap_server=0
 		;;
 		*wep*)
 			auth_type=wep
