diff -Nur a/interface-ip.c netifd-2019-08-05-5e02f944/interface-ip.c
--- a/interface-ip.c	2020-05-03 17:18:38.009128593 -0700
+++ netifd-2019-08-05-5e02f944/interface-ip.c	2020-05-03 17:20:22.664628733 -0700
@@ -1612,8 +1612,6 @@
 		set_ip_lo_policy(enabled, true, ip->iface);
 		set_ip_lo_policy(enabled, false, ip->iface);
 
-		set_ip_source_policy(enabled, true, IPRULE_PRIORITY_REJECT + ip->iface->l3_dev.dev->ifindex,
-			NULL, 0, 0, ip->iface, "failed_policy", true);
 		ip->iface->policy_rules_set = enabled;
 	}
 }
