diff -Nur a/scripts/netifd-wireless.sh netifd-2019-08-05-5e02f944/scripts/netifd-wireless.sh
--- a/scripts/netifd-wireless.sh	2020-05-04 13:21:10.706128891 -0700
+++ netifd-2019-08-05-5e02f944/scripts/netifd-wireless.sh	2020-05-04 13:23:52.321399854 -0700
@@ -39,7 +39,7 @@
 }
 
 _wdev_prepare_channel() {
-	json_get_vars channel hwmode
+	json_get_vars channel hwmode cfreq2
 
 	auto_channel=0
 	enable_ht=0
@@ -329,7 +329,7 @@
 }
 
 _wdev_common_device_config() {
-	config_add_string channel hwmode htmode noscan
+	config_add_string channel hwmode htmode noscan cfreq2
 }
 
 _wdev_common_iface_config() {
