diff -Nur netifd-copy/system-linux.c netifd-2021-01-09-753c351b/system-linux.c
--- netifd-copy/system-linux.c	2021-07-19 18:54:43.163143412 -0700
+++ netifd-2021-01-09-753c351b/system-linux.c	2021-07-19 18:56:23.355608327 -0700
@@ -2675,7 +2675,7 @@
 
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(4,4,0)
 		unsigned int draft03 = 0;
-		if ((cur = tb[IPIP6_DATA_DRAFT03])) {
+		if ((cur = tb_data[IPIP6_DATA_DRAFT03])) {
 			draft03 = blobmsg_get_u32(cur);
 			if (ttl > 255)
 				return -EINVAL;
