--- a/art/AnwiDriverInterface.h
+++ b/art/AnwiDriverInterface.h
@@ -10,7 +10,10 @@
 #ifdef __LINUX_MIPS64_ARCH__
 #include <linux/version.h>
 #endif
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(3,2,0)
+
+#ifdef __KERNEL__
+#include <linux/version.h>
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(3,2,0))
 
 #if defined(THIRD_PARTY_INTEL)
 #if !defined(__LINUX_MIPS64_ARCH__) && !defined(__LINUX_ARM_ARCH__)&& !defined(__i386__)
@@ -22,6 +25,7 @@
 #endif
 #endif
 #endif
+#endif
 #endif
 
 extern ANWIDLLSPEC int AnwiDriverDetach(void);
--- a/common/hw_routines.c
+++ b/common/hw_routines.c
@@ -8,7 +8,7 @@
 
 #if defined(Linux) || defined(__APPLE__)
 #include <unistd.h>
-#include <sys/types.h>
+ //#include <sys/types.h>
 #include <sys/stat.h>
 #include <sys/time.h>
 #include <sys/mman.h>
--- a/devlib/qc98xx/Qc98xxEepromPrint.c
+++ b/devlib/qc98xx/Qc98xxEepromPrint.c
@@ -6,6 +6,7 @@
 #include <string.h>
 #include <math.h>
 #include <ctype.h>
+#include <sys/types.h>
 
 #include "wlantype.h"
 #include "smatch.h"
--- a/devlib/qc98xx/Qc98xxEepromSave.h
+++ b/devlib/qc98xx/Qc98xxEepromSave.h
@@ -1,6 +1,7 @@
 #ifndef _QC98XX_EEPROM_SAVE_H_
 #define _QC98XX_EEPROM_SAVE_H_
 
+#include <sys/types.h>
 
 #if defined (AP_BUILD) || defined (MDK_AP)
 #define SWAP16(_x) ( (u_int16_t)( (((const u_int8_t *)(&_x))[0] ) |\
--- a/libtlv/libtlvutil/genTxBinCmdTlv.c
+++ b/libtlv/libtlvutil/genTxBinCmdTlv.c
@@ -1,6 +1,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include <sys/types.h>
 
 #if defined(_HOST_SIM_TESTING)
 #include "otaHostCommon.h"
--- a/common/linux_ansi.h
+++ b/common/linux_ansi.h
@@ -5,7 +5,6 @@
 extern "C" {
 #endif // __cplusplus 
 
-#include <sys/types.h>
 
 extern void ansi_init();
 extern void itoa(int,char *,int);
--- a/devlib/qc98xx/Qc98xxEepromSave.c
+++ b/devlib/qc98xx/Qc98xxEepromSave.c
@@ -5,6 +5,7 @@
 #include <stdlib.h>
 #include <string.h>
 #include <fcntl.h>
+#include <sys/types.h>
 
 #if 0
 #include <sys/file.h>
