Index: LinuxART2CS10.2v4.9.437/art/makefile.dart
===================================================================
--- LinuxART2CS10.2v4.9.437.orig/art/makefile.dart	2016-11-24 17:40:31.000000000 +0530
+++ LinuxART2CS10.2v4.9.437/art/makefile.dart	2017-04-12 16:19:15.857810084 +0530
@@ -54,7 +54,7 @@
 
 LDFLAGS = -L./ -L../shared/Linux -L../field/Linux -L../anwi/Linux -L../Link/Linux -L../libtlv/outputTemplate/Linux -L../libtlv/libtlvutil/Linux -L../LinkQc9K/Linux -L../calibration/cal-2p/Linux_ar9300 -L../devlib/Linux_ar9287 -L../devlib/Linux_qc98xx -L../devlib/Linux_ar9300 \
 -rdynamic \
--lm -lpthread -ldl \
+-lm -lpthread -ldl -lrt \
 -lfield -lpart -lanwi -llinkAr9k -ltlvtemplate -ltlvutil -lLinkQc9K -lcal-2p -lar9300 -lar9287 -lqc98xx\
 
 BIN = $(OUTPUT_DIR)/nart.out
