--- a/CMakeLists.txt	2019-11-09 08:32:36.901890117 +0100
+++ b/CMakeLists.txt	2019-11-09 08:35:11.054918685 +0100
@@ -32,8 +32,8 @@
 include_directories(interface/vcos/pthreads)
 include_directories(interface/vmcs_host/linux)
 include_directories(interface/vmcs_host)
-include_directories(interface/vmcs_host/khronos)
-include_directories(interface/khronos/include)
+#include_directories(interface/vmcs_host/khronos)
+#include_directories(interface/khronos/include)
 include_directories(${PROJECT_BINARY_DIR})
 include_directories(interface/vchiq_arm)
 #include_directories(tools/inet_transport)
@@ -66,9 +66,9 @@
 add_subdirectory(interface/vcos)
 add_subdirectory(interface/vmcs_host)
 add_subdirectory(interface/vchiq_arm)
-if(NOT ARM64)
-   add_subdirectory(interface/khronos)
-endif()
+#if(NOT ARM64)
+#   add_subdirectory(interface/khronos)
+#endif()
 
 #add_subdirectory(opensrc/tools/lua)
 if(BUILD_MMAL)
@@ -82,9 +82,9 @@
 add_subdirectory(host_applications/android/apps/vidtex)
 endif(BUILD_MMAL_APPS)
 
-if(NOT ARM64)
-   add_subdirectory(middleware/openmaxil)
-endif()
+#if(NOT ARM64)
+#   add_subdirectory(middleware/openmaxil)
+#endif()
 
 # 3d demo code
 #if(NOT ANDROID)
