diff --git a/CMakeLists.txt b/CMakeLists.txt
index fe67fc8..fa2d436 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -8,13 +8,9 @@ SET(PROJECT_VER_PATCH 0)
 SET(PROJECT_VER "${PROJECT_VER_MAJOR}.${PROJECT_VER_MINOR}.${PROJECT_VER_PATCH}")
 SET(PROJECT_APIVER "${PROJECT_VER}")
 
-if(ARM64)
-   set(BUILD_MMAL FALSE)
-   set(BUILD_MMAL_APPS FALSE)
-else()
-   set(BUILD_MMAL TRUE)
-   set(BUILD_MMAL_APPS TRUE)
-endif()
+set(BUILD_MMAL FALSE)
+set(BUILD_MMAL_APPS FALSE)
+
 set(vmcs_root ${PROJECT_SOURCE_DIR})
 get_filename_component(VIDEOCORE_ROOT . ABSOLUTE)
 
@@ -71,11 +67,11 @@ if(NOT ARM64)
 endif()
 
 #add_subdirectory(opensrc/tools/lua)
-if(BUILD_MMAL)
-   include_directories(interface/mmal)
-   add_subdirectory(interface/mmal)
-   add_subdirectory(containers)
-endif()
+#if(BUILD_MMAL)
+#   include_directories(interface/mmal)
+#   add_subdirectory(interface/mmal)
+#   add_subdirectory(containers)
+#endif()
 
 # VidTex supports Android and Linux
 if(BUILD_MMAL_APPS)
