--- a/src/configuration.c	2020-04-14 21:31:22.057292953 +0200
+++ b/src/configuration.c	2020-04-14 21:32:13.712364449 +0200
@@ -35,10 +35,10 @@
  */
 //@{
 /// Underlying string concatenation macro.
-#define MPTCPD_CONCAT_IMPL(x, ...) x ## __VA_ARGS__
+//#define MPTCPD_CONCAT_IMPL(x, ...) x ## __VA_ARGS__
 
 /// Concatenate strings using the preprocessor.
-#define MPTCPD_CONCAT(x, ...) MPTCPD_CONCAT_IMPL(x, __VA_ARGS__)
+//#define MPTCPD_CONCAT(x, ...) MPTCPD_CONCAT_IMPL(x, __VA_ARGS__)
 //@}
 
 // Compile-time default logging choice
@@ -47,7 +47,7 @@
 # error Problem configuring default log message destination.
 #endif
 /// Name of the default logging function determined at compile-time.
-#define MPTCPD_SET_LOG_FUNCTION MPTCPD_CONCAT(l_log_set_, MPTCPD_LOGGER)
+#define MPTCPD_SET_LOG_FUNCTION l_log_set_stderr
 
 /**
  * @brief Get the function that sets the log message destination.
