--- a/rpi-eeprom-update	2020-08-23 10:36:21.892002253 +0200
+++ b/rpi-eeprom-update	2020-08-23 10:36:41.819672745 +0200
@@ -166,7 +166,7 @@
                 || die "Failed to copy ${TMP_EEPROM_IMAGE} to ${BOOTFS}"
 
         # For NFS mounts ensure that the files are readable to the TFTP user
-        chmod -f go+r "${BOOTFS}/pieeprom.upd" "${BOOTFS}/pieeprom.sig"
+        #chmod -f go+r "${BOOTFS}/pieeprom.upd" "${BOOTFS}/pieeprom.sig"
    fi
 
    if [ -n "${VL805_UPDATE_IMAGE}" ]; then
@@ -175,7 +175,7 @@
         cp -f "${VL805_UPDATE_IMAGE}" "${BOOTFS}/vl805.bin"
 
         # For NFS mounts ensure that the files are readable to the TFTP user
-        chmod -f go+r "${BOOTFS}/vl805.bin" "${BOOTFS}/vl805.sig"
+        #chmod -f go+r "${BOOTFS}/vl805.bin" "${BOOTFS}/vl805.sig"
    fi
 
    cp -f "${RECOVERY_BIN}" "${BOOTFS}/recovery.bin" \
