diff --git a/exthost/Linux/app/diag_demo/build/Makefile b/exthost/Linux/app/diag_demo/build/Makefile
index 12f4534..a5b7348 100755
--- a/exthost/Linux/app/diag_demo/build/Makefile
+++ b/exthost/Linux/app/diag_demo/build/Makefile
@@ -70,7 +70,7 @@ all: wpan_demo
 .PHONY:
 wpan_demo: $(OBJS)
 	$(CC) $(CFLAGS) $(OBJS)  -o diag_demo $(LDFLAGS)
-
+	$(CC) $(CFLAGS) -shared -fPIC $(DIAG_DIR)/diag_api.c  -o libdiag_demo.so $(LDFLAGS)
 .PHONY:
 clean:
 	rm -rf $(OBJS)
