diff --git a/exthost/Linux/daemon/mml/mml.c b/exthost/Linux/daemon/mml/mml.c
index 30392dc..8792056 100755
--- a/exthost/Linux/daemon/mml/mml.c
+++ b/exthost/Linux/daemon/mml/mml.c
@@ -103,6 +103,11 @@ int mml_open_q(void* cxt, uint8_t qid, uint8_t dir, sem_t* sem)
 {
     MML_CXT_T* mml_cxt = GET_MML_CXT(cxt);
 
+	if(!mml_cxt) {
+		IOTD_LOG(LOG_TYPE_CRIT,"MML Open: invalid mml ctx\n");
+		return IOTD_ERROR;
+	}
+
     if(qid >= IOTD_MAX_NUM_Q){
         IOTD_LOG(LOG_TYPE_CRIT,"MML Open: invalid q id\n");
         return IOTD_ERROR;
